### PR TITLE
Refactored clang version check into a python script.

### LINKED ISSUE #1470: check clang-version on Windows
pre-commit calls `ci/cmake-format-all.sh` to check the version. However, it doesn't work on Windows.

### PR DESCRIPTION
**Pull request by Dimi1010**
Fixes #1470 
The PR changes the clang version check script from a bash script to a python script. The idea is that since pre-commit is a python package, the system running it will (hopefully) have a system-wide python interpreter installed and can run the checker script regardless of the OS.

### CODE REVIEW

File: ci/check-clang-format-version.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import subprocess
+
+EXPECTED_CLANG_VERSION = "18.1.6"
+
+
+def main():
+    result = subprocess.run("clang-format --version", capture_output=True)
```
Comment (tigercosmos) [inline ci/check-clang-format-version.py:7]: `subprocess.run(["clang-format", "--version"]...`

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -5,8 +5,8 @@ repos:
     hooks:
       - id: check-clang-format-version
         name: Check clang-format version
-        entry: ./ci/check-clang-format-version.sh
-        language: script
+        entry: python ./ci/check-clang-format-version.py
```
Comment (tigercosmos) [inline .pre-commit-config.yaml:8]: explicitly use Python3?

Comment (Dimi1010) [reply .pre-commit-config.yaml:8]: Sure. `os.run` is not supported pre python 3.6 anyway.

File: ci/check-clang-format-version.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    result.check_returncode()
+
+    version_str = result.stdout.decode("utf-8").split(" ")[2].strip()
+    if version_str != EXPECTED_CLANG_VERSION:
+        print(
+            f"Error: Found clang-format version {version_str}, but {EXPECTED_CLANG_VERSION} is required."
+        )
+        exit(1)
```
Comment (seladb) [inline ci/check-clang-format-version.py:15]: In python the convention is to throw an exception. It's exit value is `1`:
```python
raise ValueError(f"Error: Found clang-format version {version_str}, but {EXPECTED_CLANG_VERSION} is required.")
```

File: ci/check-clang-format-version.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+    print("Clang format version satisfied.")
+    exit(0)
```
Comment (seladb) [inline ci/check-clang-format-version.py:18]: This line is not needed. The default exit value is `0`

Comment (Dimi1010) [reply ci/check-clang-format-version.py]: Fixed c0b4713

File: ci/check-clang-format-version.py (hunk 3 above)
Comment (Dimi1010) [reply ci/check-clang-format-version.py]: Fixed c0b4713

File: ci/check-clang-format-version.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import subprocess
+
+EXPECTED_CLANG_VERSION = "18.1.6"
+
+
+def main():
+    result = subprocess.run(("clang-format", "--version"), capture_output=True)
+    result.check_returncode()
+
+    version_str = result.stdout.decode("utf-8").split(" ")[2].strip()
```
Comment (seladb) [inline ci/check-clang-format-version.py:10]: nit: the default split character is space, so you can just do: `.split()`

File: ci/check-clang-format-version.py (hunk 1 above)
Comment (Dimi1010) [reply ci/check-clang-format-version.py]: Fixed c6efe07

**Review by seladb — APPROVED** [review]
Small nit, otherwise LGTM

File: ci/check-clang-format-version.py (hunk 5 above)
Comment (Dimi1010) [reply ci/check-clang-format-version.py]: Thanks. Updated in de4d3d8

**Pull request merged**